### PR TITLE
Add docs for feature Extend CSI Snapshot support to Longhorn snapshot

### PR DESCRIPTION
**Pull request by PhanLe1010**
longhorn/longhorn#2534

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@ package-lock.json
 *.tern-port
 */**/.tern-port
 .DS_Store
+.idea
```
Comment (jenting) [inline .gitignore:8]: This one is not necessary.

Comment (PhanLe1010) [reply .gitignore:8]: Oh. Could you explain why?

Comment (jenting) [reply .gitignore:8]: Oh, you use Golang IDE, right?

Comment (PhanLe1010) [reply .gitignore:8]: Yeah, git keeps showing .idea file for me. So I added this 😂

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+  - CSI VolumeSnapshot associated with Longhorn snapshot is available since v1.3.0
 
-Available since v1.1.0
+## Overview
+
+In Kubernetes, a [CSI VolumeSnapshot](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) represents a snapshot of a volume on a storage system.
+When you create a CSI VolumeSnapshot, the storage system will take a snapshot of the corresponding volume.
+You can then use the CSI VolumeSnapshot to create a new volume that has the same content as the content of the CSI VolumeSnapshot.
+
+In Longhorn, there are 2 concepts related to the content of a volume at particular moment: [Longhorn backup](../../concepts/#3-backups-and-secondary-storage) and [Longhorn snapshot](../../concepts/#24-snapshots).
```
Comment (jenting) [inline content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md:24]: Suggest describes snapshot first then backup.

```suggestion
In Longhorn, there are 2 concepts related to the content of a volume at a particular moment: [Longhorn snapshot](../../concepts/#24-snapshots) and [Longhorn backup](../../concepts/#3-backups-and-secondary-storage).
```

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+Longhorn snapshot is the content of a Longhorn volume at a particular moment. It is stored inside the cluster.
+A Longhorn backup is associated to a Longhorn snapshot, but it is stored externally inside a [backup target](../backup-and-restore/set-backup-target/).
```
Comment (jenting) [inline content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md:26]:
```suggestion
A Longhorn backup is associated with a Longhorn snapshot, but it is stored externally inside a [backup target](../backup-and-restore/set-backup-target/).
```

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md [hunk 4]
```diff
@@ -1,11 +1,30 @@
...
+
+When you create a CSI VolumeSnapshot, you can specify whether the CSI VolumeSnapshot should be associated to a Longhorn snapshot (content is stored inside cluster) or a Longhorn backup (content is stored both inside cluster and uploaded to external backup target).
```
Comment (jenting) [inline content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md:28]:
```suggestion
When you create a CSI VolumeSnapshot, you can specify whether the CSI VolumeSnapshot should be associated with a Longhorn snapshot (content is stored inside cluster) or a Longhorn backup (content is stored both inside the cluster and uploaded to external backup target).
```

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md [hunk 5]
```diff
@@ -1,11 +1,30 @@
...
+
+After install the [prerequisite](./enable-csi-snapshot-support), you can see [CSI VolumeSnapshot Associated with Longhorn Backup](./csi-volume-snapshot-associated-with-longhorn-backup) and [CSI VolumeSnapshot Associated with Longhorn Snapshot](./csi-volume-snapshot-associated-with-longhorn-snapshot) for more detail about how to create/restore a CSI VolumeSnapshot and the perameters you can set.
```
Comment (jenting) [inline content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md:30]: Suggest describes snapshot first then backup.

```suggestion
After installing the [prerequisite](./enable-csi-snapshot-support), you can see [CSI VolumeSnapshot Associated with Longhorn Snapshot](./csi-volume-snapshot-associated-with-longhorn-snapshot) and [CSI VolumeSnapshot Associated with Longhorn Backup](./csi-volume-snapshot-associated-with-longhorn-backup) for more detail about how to create/restore a CSI VolumeSnapshot and the parameters you can set.
```

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-snapshot.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
+---
+title: CSI VolumeSnapshot Associated with Longhorn Snapshot
+weight: 3
```
Comment (jenting) [inline content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-snapshot.md:3]: Can we change this to weight 2?
I'd prefer to describe snapshot first then backup.

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
+---
+title: CSI VolumeSnapshot Associated with Longhorn Backup
+weight: 2
```
Comment (jenting) [inline content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md:3]: Can we change this to weight 3?
I'd prefer to describe snapshot first then backup.

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+  source:
+    volumeSnapshotContentName: test-existing-backup
+```
+
+Now you can create a `PerstistantVolumeClaim` object that refers to the newly created `VolumeSnapshot` object.
+For an example see [Restore PVC from CSI VolumeSnapshot Associated With Longhorn Backup](#restore-pvc-from-csi-volumesnapshot-associated-with-longhorn-backup) above.
+
+## Current Limitation
+
+* Longhorn volume must be in attached state to create/restore snapshot using CSI VolumeSnapshot mechanism.
```
Comment (jenting) [inline content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md:143]:
```suggestion
* Longhorn volume must be in an attached state to create/restore snapshot using CSI VolumeSnapshot mechanism.
```

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-snapshot.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 5Gi
+```
+Note that the `spec.resources.requests.storage` value must be the same as the size of `VolumeSnapshot` object.
+
+
+## Current Limitation
+* Longhorn volume must be in attached state to create/restore snapshot using CSI VolumeSnapshot mechanism.
```
[the remark this answers is not in the document]
Comment (PhanLe1010) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-snapshot.md]: Thank you. Fixed

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md (hunk 2 above)
Comment (PhanLe1010) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md]: Thank you. Fixed

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md (hunk 4 above)
Comment (PhanLe1010) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md]: Thank you. Fixed

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md (hunk 5 above)
Comment (PhanLe1010) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md]: Thank you. Fixed

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md (hunk 7 above)
Comment (PhanLe1010) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md]: Thank you. Fixed

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-snapshot.md (hunk 6 above)
Comment (PhanLe1010) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-snapshot.md]: Thank you. Fixed

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md (hunk 8 above)
Comment (PhanLe1010) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md]: "an attached state" sounds like there are multiple attached state. Should we use "the attached" state instead?

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-snapshot.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (PhanLe1010) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-snapshot.md]: Same as above

File: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md (hunk 8 above)
Comment (jenting) [reply content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backup.md]: Sound good to me :smiley:

**Review by jenting — APPROVED** [review]
LGTM

**Review by innobead — APPROVED** [review]

**Pull request merged**